lib/auth/backends/ChainBackend.js

```diff
@@ -0,0 +1,188 @@
+                message: {
+                    body: ChainBackend._mergeObjects(res),
+                },
+            });
+        });
+    }
+
+    getEmailAddresses(canonicalIDs, options, callback) {
+        this._forEachClient((client, done) => {
+            if (typeof client.getEmailAddresses !== 'function') {
+                return done(null, { message: { body: {} } });
+            }
+            return client.getEmailAddresses(canonicalIDs, options, done);
+        }, (err, res) => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, {
+                message: {
+                    body: ChainBackend._mergeObjects(res),
+                },
+            });
+        });
+    }
+
+    /*
+     * merge policy responses into a single message
+     */
+    static _mergePolicies(policyResponses) {
+        const policyMap = {};
+
+        policyResponses.forEach(resp => {
+            if (!resp.message || !Array.isArray(resp.message.body)) {
+                return;
+            }
+
+            resp.message.body.forEach(policy => {
+                const arn = policy.arn || '';
+                if (!policyMap[arn]) {
+                    policyMap[arn] = policy.isAllowed;
+                }
+                // else is duplicate policy
+            });
+        });
+
+        const policyList = Object.keys(policyMap).map(arn => {
+            if (arn === '') {
+                return { isAllowed: policyMap[arn] };
+            }
+            return { isAllowed: policyMap[arn], arn };
+        });
+
+        return policyList;
+    }
+
+    /*
+        response format:
+            {  message: {
+                body: [{}],
+                code: number,
+                message: string,
+            } }
+     */
+    checkPolicies(requestContextParams, userArn, options, callback) {
+        this._forEachClient((client, done) => {
+            if (typeof client.checkPolicies !== 'function') {
+                return done(null, { message: { body: [] } });
+            }
+            return client.checkPolicies(
+                requestContextParams,
+                userArn,
+                options,
+                done
+            );
+        }, (err, res) => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, {
+                message: {
+                    body: ChainBackend._mergePolicies(res),
+                },
+            });
+        });
+    }
+
+    healthcheck(reqUid, callback) {
+        this._forEachClient((client, done) => {
+            if (typeof client.healthcheck !== 'function') {
+                return done(null, { code: 200, message: 'OK' });
+            }
+            return client.healthcheck(reqUid, done);
+        }, (err, res) => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, res);
```

Comment:
Seems fine in that if any backend fails the check, an error will be returned, but doesn't this change the shape of `res`?

Reply:
after taking a step back and having a better look at this, res is even more important in the case of an error.
What I had in mind was to just returning the error in the case of that  the healthcheck fails